tgui/packages/tgui/interfaces/Safe.js

```diff
@@ -0,0 +1,119 @@
+import { useBackend } from "../backend";
+import { Window } from "../layouts";
+import { Box, Button, Flex, Section } from "../components";
+const safeDialSrc = require('../assets/safe_dial.png');
+
+export const SafeContents = (properties, context) => {
+  const { data, act } = useBackend(context);
+  const {
+    contents,
+  } = data;
+
+  const close = () => {
+    act('open');
+  };
+
+  return (
+    <Section>
+      <Button content="Close" onClick={close} />
+      <Section>
+        {contents.map((content, i) => {
+          const {
+            name, index, sprite,
+          } = content;
+          const retrieve = () => {
+            act('retrieve', { index });
+          };
+          const src = `${sprite}.png`;
+          return (
+            <Flex alignItems="center" key={i}>
+              <img src={src} />
+              <Button content={name} onClick={retrieve} />
+            </Flex>
+          );
+        })}
+      </Section>
+    </Section>
+  );
+};
+
+export const SafeLock = (properties, context) => {
+  const { data, act } = useBackend(context);
+  const {
+    dial, rotation, locked,
+  } = data;
+  const open = () => {
+    act('open');
+  };
+  const turn = (direction, amount) => {
+    act('turn', { amount, direction });
+  };
+
+  return (
+    <Section>
+
+      <Flex alignItems="stretch" flexDirection="column">
+
+        <Flex alignItems="center" flexDirection="column">
+          <Button content="Open" onClick={open} />
+
+        </Flex>
+
+        <Flex justifyContent="space-between" alignItems="center">
+          <Flex>
+
+            <Button content="Left 50" onClick={() => turn('left', 50)} icon="arrow-left" />
+            <Button content="Left 10" onClick={() => turn('left', 10)} icon="arrow-left" />
+            <Button content="Left 1" onClick={() => turn('left', 1)} icon="arrow-left" />
+          </Flex>
+
+          <Flex flexGrow={1} alignItems="center" flexDirection="column">
+            {dial}
+          </Flex>
+
+          <Flex>
+
+            <Button disabled={!locked} content="Right 1" onClick={() => turn('right', 1)} icon="arrow-left" />
```

Comment:
This makes the "Right" icon with left-pointing arrow, which is very odd
![firefox-20201026-093444](https://user-images.githubusercontent.com/7831163/97156378-87f8c100-176e-11eb-8ab8-91ac49ac5a07.png)

I assume this is done because `-> Right` looks weird.

If you want to have icon *following* the text (so `Right ->`) you need to explicitly declare button contents, i.e.:
```jsx
<Button disabled={!locked} onClick={() => turn('right', 1)}>Right 1 <Icon name="arrow-right"/></Button>
```